app/config.c

```diff
@@ -493,6 +499,103 @@ parse_config_from_file(uint8_t port_id, char *q_arg)
 	return 0;
 }
 
+static int rate_limit_ipv4(uint32_t naddr, uint32_t num)
+{
+	union rlimit_addr *addr;
+	uint8_t range_id;
+	static uint8_t next_range_id = 0;
+
+	addr = (union rlimit_addr *)&naddr;
+	range_id = rlimit4_lookup_table[addr->network];
+	// check if this cidr range is the lookup table
+	if (range_id == INVALID_RLIMIT_RANGE) {
+		range_id = next_range_id++;
+
+		if (range_id >= MAX_RLIMIT_RANGE) { // if not found
+			return -1;
+		}
+	}
+
+	// set slot for this cidr range in the lookup table
+	// and set the max packet rate for this dest addr
+	rlimit4_lookup_table[addr->network] = range_id;
+	rlimit4_max[range_id][addr->host] = num;
+
+	return 0;
+}
+
+int rate_limit_address(cmdline_ipaddr_t *ip, uint32_t num);
+int rate_limit_address(cmdline_ipaddr_t *ip, uint32_t num)
+{
+	uint8_t next_hop;
+	int i;
+	uint32_t netmask, netaddr, j;
+
+	if (ip->prefixlen > 0) {
+		// rate limit range
+		if (ip->family == AF_INET) {
+			netmask = ((1 << ip->prefixlen) - 1) << (32 - ip->prefixlen);
+			netaddr = rte_be_to_cpu_32(ip->addr.ipv4.s_addr) & netmask;
+			for (j = netaddr; j < netaddr + (1 << (32 - ip->prefixlen)); j++) {
```

Comment:
may you put `netaddr + (1 << (32 - ip->prefixlen))` in a var outside the loop ? its easier to debug/print